docs/debug/mpcie_ethernet.md

```diff
@@ -0,0 +1,722 @@
+```
+
+Pinging `eth1` interface address from different PC works well:
+
+```
+ping 192.168.0.120
+```
+
+output:
+
+```
+PING 192.168.0.120 (192.168.0.120) 56(84) bytes of data.
+64 bytes from 192.168.0.120: icmp_seq=1 ttl=64 time=0.755 ms
+64 bytes from 192.168.0.120: icmp_seq=2 ttl=64 time=0.306 ms
+```
+
+* Second interface (00:e0:4c:68:05:c8 MAC address) from extension card:
+
+> Reboot was executed first.
+
+```
+ifconfig eth0 up
+```
+
+output:
+
+```
+[   63.775838] r8169 0000:03:00.0: firmware: direct-loading firmware rtl_nic/rtl8168e-3.fw
+[   63.880564] r8169 0000:03:00.0 eth0: link down
+[   63.885105] r8169 0000:03:00.0 eth0: link down
+[   63.885192] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
+[   65.798066] r8169 0000:03:00.0 eth0: link up
+[   65.802396] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
+```
+
+```
+dhclient  eth0
+ifconfig eth0
+```
+
+output:
+
+```
+eth0      Link encap:Ethernet  HWaddr 00:e0:4c:68:05:c8
+          inet addr:192.168.0.123  Bcast:192.168.0.255  Mask:255.255.255.0
+          inet6 addr: fe80::2e0:4cff:fe68:5c8/64 Scope:Link
+          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
+          RX packets:4 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:10 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:1000
+          RX bytes:1300 (1.2 KiB)  TX bytes:1332 (1.3 KiB)
+```
+
+Pinging some internal or external adresses from APU2 works:
+
+```
+ping 192.168.0.1
+```
+
+output:
+
+```
+PING 192.168.0.1 (192.168.0.1) 56(84) bytes of data.
+64 bytes from 192.168.0.1: icmp_seq=1 ttl=64 time=0.750 ms
+64 bytes from 192.168.0.1: icmp_seq=2 ttl=64 time=0.266 ms
+```
+
+```
+ping google.pl
+```
+
+output:
+
+```
+PING google.pl (172.217.20.195) 56(84) bytes of data.
+64 bytes from waw02s08-in-f195.1e100.net (172.217.20.195): icmp_seq=1 ttl=57 time=6.87 ms
+64 bytes from waw02s08-in-f195.1e100.net (172.217.20.195): icmp_seq=2 ttl=57 time=6.75 ms
+```
+
+Pinging `eth0` interface address from different PC works well:
+
+```
+ping 192.168.0.123
+```
+
+output:
+
+```
+PING 192.168.0.123 (192.168.0.123) 56(84) bytes of data.
+64 bytes from 192.168.0.123: icmp_seq=1 ttl=64 time=0.800 ms
+64 bytes from 192.168.0.123: icmp_seq=2 ttl=64 time=0.313 ms
+```
+
+### Conclusion
+
+* Ethernet interfaces from extension card inserted into mPCIe slot 1 operate
+  properly in terms of device enumeration and DHCP lease.
+
+* Ethernet interfaces from extension card inserted into mPCIe slot 2 do not
+  operate properly. They are not enumerated by kernel.
+
+* After forcing mPCIe clock to be always on, Ethernet interfaces from extension
+  card inserted into mPCIe 2 slot operate in the same fashion as when inserted
+  into mPCIe slot 1 (when it comes to device enumeration and DHCP lease, at
+  least). Note that
+  [corresponding change](https://github.com/pcengines/coreboot/blob/v4.5.8/src/mainboard/pcengines/apu2/romstage.c#L95)
+  exists for mPCIe slot 1.
```

Comment:
@macpijan then I think we need that change in coming release. Can you try operate more interfaces at once ? And also try to obtain IP address on more then one port ?